### PR TITLE
217 add section chapter

### PR DESCRIPTION
**Pull request by yishain11**

### CODE REVIEW

File: frontend/src/pages/Agreement.tsx [hunk 1]
```diff
@@ -133,12 +135,37 @@ const Agreement: FC = () => {
...
+                  <Divider
+                    className="divider"
+                    textAlign="center"
+                    variant="fullWidth"
+                    sx={{ paddingBottom: 1 }}
+                  >
+                    <IconButton
+                      onClick={() => {
+                        setCurrentChapterId(chapter.id);
+                        setCurrentSectionIndex(1);
```
Comment (avirankatz) [inline frontend/src/pages/Agreement.tsx:147]: This should be changed to the index of the first section in the current chapter

File: frontend/src/pages/Agreement.tsx [hunk 2]
```diff
@@ -133,12 +135,37 @@ const Agreement: FC = () => {
...
+                    </IconButton>
+                  </Divider>
                   {chapter?.sections?.map((section) => (
                     <div key={section.id}>
                       <SectionCard section={section} />
                       <Divider className="divider" textAlign="center" variant="fullWidth">
                         <IconButton
                           onClick={() => {
+                            setCurrentChapterId(chapter.id);
+                            setCurrentSectionIndex(section.index);
```
Comment (avirankatz) [inline frontend/src/pages/Agreement.tsx:168]:
```suggestion
                            setCurrentSectionIndex(section.index + 1);
```

File: frontend/src/pages/Agreement.tsx [hunk 3]
```diff
@@ -179,6 +187,25 @@ const Agreement: FC = () => {
...
         </Stack>
       </Stack>
+      <TextEditorPopup
+        isOpen={isTextPopupOpen}
+        parentSection={t(StringBank.NEW_SECTION)}
+        onComplete={(editorContent) => {
+          const content = editorContent.variables.content;
+          const variables = {
+            chapterId: currentChapterId,
+            sectionIndex: currentSectionIndex + 1,
```
Comment (avirankatz) [inline frontend/src/pages/Agreement.tsx:197]:
```suggestion
            sectionIndex: currentSectionIndex,
```

**Review by avirankatz — CHANGES_REQUESTED** [review]
Good. See minor requests

File: frontend/src/pages/Agreement.tsx (hunk 2 above)
Comment (yishain11) [reply frontend/src/pages/Agreement.tsx:168]: fixed

File: frontend/src/pages/Agreement.tsx (hunk 3 above)
Comment (yishain11) [reply frontend/src/pages/Agreement.tsx:197]: changed

**Pull request merged**